tests/testcontainers/logstash.py

```diff
@@ -23,128 +28,172 @@ class LogstashContainer(DockerContainer):  # type: ignore
         with LogstashContainer() as lc:
             # NOTE: container will terminate once out of this with statement
             url = lc.get_url()
-
-    With custom config
-    ------------------
-    ::
-
-        lgc = LogstashContainer(port=self.logstash_http_port)
-        logstash_config = '''\
-            input {{
-              http {{
-                port => {logstash_http_port}
-                codec => json_lines
-              }}
-            }}
-
-            output {{ stdout {{ codec => json_lines }} }}
-            '''.format(logstash_http_port=self.logstash_http_port)
-        lgc.with_env("CONFIG_STRING", logstash_config)
     """
 
     _DEFAULT_IMAGE = "docker.elastic.co/logstash/logstash"
     _DEFAULT_VERSION = "7.16.0"
     _DEFAULT_PORT = 5044
     _DEFAULT_API_PORT = 9600
+    _DEFAULT_USERNAME = "USERNAME"
+    _DEFAULT_PASSWORD = "PASSWORD"
 
     def __init__(
         self,
         image: str = _DEFAULT_IMAGE,
         version: str = _DEFAULT_VERSION,
         port: int = _DEFAULT_PORT,
         api_port: int = _DEFAULT_API_PORT,
+        username: str = _DEFAULT_USERNAME,
+        password: str = _DEFAULT_PASSWORD,
     ):
         image = f"{image}:{version}"
         super(LogstashContainer, self).__init__(image=image)
 
         self.port = port
         self.api_port = api_port
+        self.exposed_port = 0
+        self.ssl_assert_fingerprint = ""
+
+        self.logstash_user: str = username
+        self.logstash_password: str = password
 
         self.with_exposed_ports(self.port)
         self.with_exposed_ports(self.api_port)
-        self._configure()
 
-        self._previous_message_count = 0
+        self._last_reset_message_count: int = 0
+        self._previous_message_count: int = 0
 
     def _configure(self) -> None:
         """
-        You can override any value set here by calling <instance>.with_env(...) after initializing this class
+        Values set here will override any value set by calling <instance>.with_env(...)
+        after initializing this class before <instance>.start()
         """
-        self.with_env("LOG_LEVEL", os.environ.get("LOGSTASH_LOG_LEVEL", "warn"))
+
+        # TODO: let's generate the certificate and key with code, current cert will expire Nov 17 15:04:09 2025 GMT
```

Comment:
logstash docker image misses the tool in order to generate the cert and key inside the container. I will look on how to do from code in python, and then save the output on files in the container and reference them instead